python/ray/serve/backend_state.py

```diff
@@ -1084,20 +1082,52 @@ def __init__(self, controller_name: str, detached: bool,
         self._goal_manager = goal_manager
         self._create_backend_state: Callable = lambda name: BackendState(
             name, controller_name, detached,
-            long_poll_host, goal_manager, self._checkpoint)
+            long_poll_host, goal_manager, self._save_checkpoint_func)
         self._backend_states: Dict[BackendTag, BackendState] = dict()
         self._deleted_backend_metadata: Dict[BackendTag,
                                              BackendInfo] = OrderedDict()
+        self._recover_from_checkpoint()
 
+    def _get_all_running_replicas(self):
+        # This ray call should be mocked in all unit tests
```

Comment:
If this should be mocked in unit tests, let's make it a parameter passed in (dependency inject)

Reply:
I really prefer to not mock things whenever possible

Reply:
i'm currently doing it by mocking `ray.util.list_named_actors` in test to return current mock replica instead which served its purpose, but indeed this `list_named_actors()` call did appear a few levels above where we were trying to keep everything contained in ActorReplicaWrapper. Dependency injection sounds better